alot/commands/globals.py

```diff
@@ -755,8 +755,9 @@ def apply(self, ui):
         if self.template is not None:
             # get location of tempsdir, containing msg templates
             tempdir = settings.get('template_dir')
-            tempdir = os.path.expanduser(tempdir)
-            if not tempdir:
+            if tempdir:
```

Comment:
An alternative would be to do something like:
```python
tempdir = os.path.expandvars(tempdir)
if tempdir == '/alot/templates':
    tempdir = os.path.join(os.environ['HOME'], '.config', tempdir)
````
Then you could leave XDG_CONFIG_HOME as the default path.

Reply:
I think I'm not a fan of using `expandvars`, because that would try to expand any `$` in a path, if I understand correctly. If for some reason someone uses '~/my$bill` as a path, it would expand `$bill` if the variable exists. Admittedly that's very unlikely, but why take the risk - I mean XDG_CONFIG_HOME is still used as the default path if it is defined.

But of course, I'm happy to change it if you prefer it :)

Reply:
I withdraw my comment. How about this:

```python
    tempdir = settings.get('template_dir')
    if os.path.expandvars(tempdir) == '$XDG_CONFIG_HOME/alot/template':
        tempdir = '~/.config/alot/templates'
    
    tempdir = os.path.expanduser(tempdir)
```

But I don't know, if that file path check works on platforms where the separator is different.

Reply:
I can go either way, although someone would have to go well out of their way to make `my$bill as a file or path, since shell interpolation will happen before passing the value to the file, and most gui file managers don't let you use `$` as a path, and I'm not a fan of bending over backwards to support people doing stupid things. In this case it's not a big deal either way, you could also do something like:

```python
if 'XDG_CONFIG_HOME' in os.environ:
    tempdir = os.path.expanduser(tempdir)
else:
    tempdir = os.path.join(os.environ['HOME'], '.config', 'alot', 'templates')
```

But it was just a suggestion.

Reply:
```python
    tempdir = settings.get('template_dir')
    if os.path.expandvars(tempdir) == '$XDG_CONFIG_HOME/alot/template' and 'XDG_CONFIG_HOME not in os.environ:
        tempdir = '~/.config/alot/templates'
    
    tempdir = os.path.expanduser(tempdir)
```

Reply:
My last try, inspired  by yours that adds the case where XDG_CONFIG_HOME is actually defined:

```python
    tempdir = settings.get('template_dir')
    if ('XDG_CONFIG_HOME' in os.environ
            and os.environ.get('XDG_CONFIG_HOME')):
        tempdir = tempdir.replace('$XDG_CONFIG_HOME',
                                                  os.environ.get('XDG_CONFIG_HOME'))
    elif tempdir == '$XDG_CONFIG_HOME/alot/templates':
        tempdir = '~/.config/alot/templates'
    tempdir = os.path.expanduser(tempdir)
``

Reply:
That looks fairly reasonable, although you could simplify it
```python
if os.environ.get('XDG_CONFIG_HOME', False):
     tempdir = tempdir.replace('$XDG_CONFIG_HOME', os.environ['XDG_CONFIG_HOME'])
...
```
os.environ.get allows you to avoid the initial membership test, and avoiding using after you know that it is a memeber should be slightly faster, plus if someone does something bad like remove XDG_CONFIG_HOME from the environment between the test and the assignment we get an exception instead of a race.

Reply:
That's indeed nicer. Okay, I will update the PR tomorrow.